src/setup.js

```diff
@@ -27,9 +50,62 @@ function setupMeasure(dataLayer) {
    * The DataLayerHelper to use with this application.
    * @private @const {!DataLayerHelper}
    */
-  const helper = new DataLayerHelper(dataLayer);
+  const helper = new DataLayerHelper(dataLayer, {'processNow': false});
+
+  helper.registerProcessor('config', configProcessors);
+  helper.process();
+
+  /**
+   * When called with an implementation of the eventProcessor and
+   * storageInterface API, this function registers processors to react to the
+   * 'set' and 'event' commands.
+   *
+   * @param {!ProcessorConstructor|string} eventProcessor The event
+   *     processor to use when responding to event commands
+   *     and deciding how keys should be stored, or a string
+   *     representing an object of this type.
+   * @param {!Object<string, *>} eventOptions Options objects to pass to the
+   *    event processor.
+   * @param {!StorageConstructor|string} storageInterface An
+   *     interface to a class that will handle setting
+   *     and loading long term data, or a string representing
+   *     an object of this type.
+   * @param {!Object<string, *>} storageOptions Options objects to pass to the
+   *    storage interface.
+   */
+  function configProcessors(eventProcessor, eventOptions,
+                            storageInterface, storageOptions) {
+    // TODO: Read in data from the model as extra options
+
+    let processor;
+    let storage;
+    if (typeof eventProcessor === 'string') {
+      if (!(eventProcessor in DEFAULT_EVENT_PROCESSORS)) {
+        log(`No event processor with name ${eventProcessor} found.`,
+            LogLevel.ERROR);
+        return;
+      }
+      processor =
+          new DEFAULT_EVENT_PROCESSORS[eventProcessor](eventOptions);
+    } else {
+      processor = new eventProcessor(eventOptions);
```

Comment:
I would instead define a variable "EventProcessorCtor" (or similar) and use this if/else to set it appropriately. Then afterwards just do "processor = EventProcessorCtor(eventOptions)"

Reply:
That's a great idea, done